core/src/test/scala/org/http4s/rho/RhoRoutesSuite.scala

```diff
@@ -63,7 +63,7 @@ class RhoRoutesSuite extends CatsEffectSuite with RequestRunner {
         Ok("twoparams2_" + foo + bar.getOrElse("cat"))
     }
 
-    GET / "variadic" / * |>> { tail: List[String] => Ok("route8_" + tail.mkString("/")) }
+    GET / "variadic" / * |>> { tail: List[Segment] => Ok("route8_" + tail.map(identity).mkString("/")) }
```

Comment:
I'm a bit confused as to why you need to `.map(identity)` here?

Reply:
it's just a simple way to force the use of the tail type, so if it doesn't match it will throw a runtime error

Reply:
> Thanks for the contribution, and sorry it took so long to come back to you - I'm trying to catch up with everything in this repo! :-)

no worries, we moved to 0.23 since then and the same issue exists there too

Reply:
Right - I think we should target 0.23 with this, as I'm thinking of releasing soon against series 0.23. Thoughts?

Reply:
I agree, let me change the target branch, see if there are any conflicts

Reply:
I think we should do this PR for Series/0.22, Series/0.23, and main.

Reply:
I think this is more obvious and performs the same duty.  Lets change it to this.

```suggestion
    GET / "variadic" / * |>> { tail: List[Segment] => Ok("route8_" + tail.map(_.encoded).mkString("/")) }
```

Reply:
yeah, this was just to demonstrate that picking the tail type by any means was breaking. when I was writing the test, .encoded wouldn't make much sense for proving the issue.
but .encoded going forwards will complain if the type changes again so I'll change to that